usage/test-framework/src/main/java/org/apache/brooklyn/test/framework/TestHttpCallImpl.java

```diff
@@ -66,31 +68,35 @@ public void start(Collection<? extends Location> locations) {
         }
     }
 
-    private Supplier<String> buildDataSupplier(final HttpAssertionTarget httpAssertionTarget, final String url) {
-
-        switch (httpAssertionTarget) {
+    private void doRequestAndCheckAssertions(Map<String, Duration> flags, List<Map<String, Object>> assertions,
+                                             HttpAssertionTarget target, final String url) {
+        switch (target) {
             case body:
-                return new Supplier<String>() {
+                Supplier<String> getBody = new Supplier<String>() {
```

Comment:
for checks like this which will be immediate this (anonymous class) is how we'd do it (just FYI).  if we need this in more than one or two places then I'd create a `ResourceUtils.getResourceAsStringSupplier()` to do this.

btw we normally use `ResourceUtils` for reading URL's because it will use classpath URL's and can be context aware.  not important in this case but again just FYI.